pkg/controllers/monitoring/monitoring-stack/controller.go

```diff
@@ -143,9 +141,31 @@ func (r *reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 			return ctrl.Result{RequeueAfter: 2 * time.Second}, nil
 		}
 		if err != nil {
-			return ctrl.Result{}, err
+			return r.updateStatus(ctx, req, ms, err)
 		}
 	}
+
+	return r.updateStatus(ctx, req, ms, nil)
+}
+
+func (r *reconciler) updateStatus(ctx context.Context, req ctrl.Request, ms *stack.MonitoringStack, recError error) (ctrl.Result, error) {
+	var prom monv1.Prometheus
+	key := types.NamespacedName{
+		Name:      ms.Name,
+		Namespace: ms.Namespace,
+	}
+	err := r.k8sClient.Get(ctx, key, &prom)
+	if err != nil {
+		return ctrl.Result{RequeueAfter: 2 * time.Second}, nil
```

Comment:
Should we log here something? I'm afraid that if we don't log anything here that if we enter this condition regularly we will not be able to tell that the operator is stuck. WDYT?

Reply:
I think the `controller-runtime` logs every reconciliation so such situation should be visible from the logs IMO but we can add some log statement here saying something like failing to get the Prometheus instance XYZ?

Reply:
If `controller-runtime` logs it then let's ignore my comment! I'll test it to be sure 😉

Reply:
I think @JoaoBraveCoding raises a good point. Should we ignore all errors? how about ignoring only NotFound and treat the rest as actual errors?

Reply:
Yes probably yes. Thanks. Let me update it.